File: MadeByDade.Living.React/src/bills/Bill.ts
import { z } from "zod";

export const BillSchema = z.object({
	id: z.number(),
	name: z.string(),
	amount: z.number(),
	dayDue: z.number(),
	isAutoPay: z.boolean(),
	dueType: z.enum(["Fixed", "EndOfMonth"]),
});

export const BillsSchema = z.array(BillSchema);

export type Bill = z.infer<typeof BillSchema>;

File: MadeByDade.Living.Data/Bills/Bill.cs
using System.ComponentModel.DataAnnotations;

namespace MadeByDade.Living.Data.Bills;

public class Bill
{
    public int Id { get; set; }

    [Required]
    public string Name { get; set; } = string.Empty;

    [Required]
    public decimal Amount { get; set; }

    [Required]
    public int DayDue { get; set; }

    public bool IsAutoPay { get; set; } = false;

    [Required]
    public BillDueType DueType { get; set; }
}

public enum BillDueType
{
    Fixed, EndOfMonth
}
